### PR TITLE
Track payment cancelation and rename payment flow start event

### LINKED ISSUE #6786: Track Payment Cancelation and rename payment flow start event
* `CARD_PRESENT_COLLECT_PAYMENT_TAPPED` -> `PAYMENTS_FLOW_ORDER_COLLECT_PAYMENT_TAPPED`
* `SIMPLE_PAYMENTS_FLOW_CANCELED` -> `PAYMENTS_FLOW_CANCELED` with arguments `flow=simple_payment` or `flow=order_payment` (depending where the flow started). Tracking of this event is missing on a couple of places so we have to add this

- [x] Add note to the renamed events

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #6786

### Description

* Renames `CARD_PRESENT_COLLECT_PAYMENT_TAPPED` -> `PAYMENTS_FLOW_ORDER_COLLECT_PAYMENT_TAPPED`
* Renames `SIMPLE_PAYMENTS_FLOW_CANCELED` -> `PAYMENTS_FLOW_CANCELED`
* Adds `PAYMENTS_FLOW_CANCELED` tracking when
    * Selection payment method fragment is closed during order payment flow (arg `flow=order_payment`)
    * Back click on the main simple payment fragment (arg `flow=simple_payment`)

### Testing instructions

* Use logcat with filtering by `Tracked`
* Go via Simple payments flow and Order payment flow and notice tracked events

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/simplepayments/SimplePaymentsFragmentViewModel.kt [hunk 1]
```diff
@@ -183,6 +183,10 @@ class SimplePaymentsFragmentViewModel @Inject constructor(
     }
 
     fun onBackButtonClicked() {
+        AnalyticsTracker.track(
```
Comment (AnirudhBhat) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/simplepayments/SimplePaymentsFragmentViewModel.kt:186]: ❓ What do you think about writing tests to verify this in `SimplePaymentsFragmentViewModelTests`

**Comment by AnirudhBhat** [comment]
Thank you for working on this @kidinov! I've tested the changes and it works as expected. 

I noticed a few things:
1. `PAYMENTS_FLOW_ORDER_COLLECT_PAYMENT_TAPPED` event's "country" property will be "Unknown" if fired before IPP onboarding. I believe this happens even in the production as well. 
2. When the card reader is being scanned for devices, pressing the back button will not trigger `PAYMENTS_FLOW_CANCELED` event. However, clicking on the "Cancel" button on the same dialog triggers this event. Is this expected?
3. When Simple payments flow is started we get this event - `simple_payments_flow_started`. Once we click on the "Done" button and navigate to the "TakePayment" screen, we again fire `payments_flow_canceled`. Is this expected?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/simplepayments/SimplePaymentsFragmentViewModel.kt (hunk 1 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/simplepayments/SimplePaymentsFragmentViewModel.kt:186]: In order to test it I need to refactor that which probably is not the goal of the PR?

Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/simplepayments/SimplePaymentsFragmentViewModel.kt:186]: Ahh! Didn't notice it was `AnalyticsTracker`. Completely fine! Let's not do it in this PR 👍

**Comment by kidinov** [comment]
@AnirudhBhat thank you for the review and valuable feedback 🙏

> PAYMENTS_FLOW_ORDER_COLLECT_PAYMENT_TAPPED event's "country" property will be "Unknown" if fired before IPP onboarding. I believe this happens even in the production as well.

Yep, Josh also pointed to this! I removed all extra "card reader" related fields from this event [37987a4](https://github.com/woocommerce/woocommerce-android/pull/6788/commits/37987a4f096e9b4306c39d3d3cf358c504d70f30)

> When the card reader is being scanned for devices, pressing the back button will not trigger PAYMENTS_FLOW_CANCELED event. However, clicking on the "Cancel" button on the same dialog triggers this event. Is this expected?

Is this about `payments_flow_failed` event?  I think this is a bug that was there for quite some time. Created [an issue](https://github.com/woocommerce/woocommerce-android/issues/6794)

> When Simple payments flow is started we get this event - simple_payments_flow_started. Once we click on the "Done" button and navigate to the "TakePayment" screen, we again fire payments_flow_canceled. Is this expected?

That's me being stupid 🤦‍♂️ . Thank you for catching this! Fixed here - 628885124b93b869ef12ca17a61cb33859d24120

**Review by AnirudhBhat — APPROVED** [review]
Thank you for all the changes! LGTM! :shipit:

**Pull request merged**